src/consensus/tx_verify.cpp

```diff
@@ -27,6 +29,43 @@ bool IsFinalTx(const CBaseTransaction &tx, int nBlockHeight, int64_t nBlockTime)
     return true;
 }
 
+
+bool IsLicenseTx(const CBaseTransaction& tx) {
+	if (tx.IsNull() || tx.IsCoinBase())
+		return false;
+
+	if (tx.vin.size() != 1)
+		return false;
+
+	auto ComesFromWDMO = [&](CTxIn vin, CScript wdmoScript) {
+		COutPoint prevout = vin.prevout;
+		CBaseTransactionRef txOut;
+		uint256 hashBlock;
+
+		if (GetTransaction(prevout.hash, txOut, Params().GetConsensus(), hashBlock))
+			for (const auto& vout : txOut.get()->vout)
+				if (vout.scriptPubKey == wdmoScript)
+					return true;
```

Comment:
It would be a major security issue if we left it like this. A transaction pointed to by `vin.prevout` can serve outputs to multiple recipients. Any recipient having outputs in this transaction would be able to issue a valid LTX.
We need to check only the output which number (`n`) is equal to `vin.vout`

Reply:
Done